test/e2e/plugin_lifecycle/plugin_lifecycle_test.go

```diff
@@ -5,39 +5,209 @@
+		// Test case: clean plugins if any installed already
+		It("clean plugins if any installed already", func() {
+			err := tf.PluginCmd.CleanPlugins()
+			Expect(err).To(BeNil(), "should not get any error for plugin clean")
+			pluginsList, err := tf.PluginCmd.ListPlugins()
+			Expect(err).To(BeNil(), "should not get any error for plugin list")
+			Expect(len(pluginsList)).Should(BeNumerically("==", 0), "plugins list should not return any plugins after plugin clean")
 		})
-		// Test case: List plugins and make sure all plugins added in setup node should be listed
-		It("list and validate plugins", func() {
-			ListAndValidatePlugins(tf, discoveryMap)
+		// Test case: install all plugins from framework.PluginsForLifeCycleTests, and validate the installation by running describe command on each plugin
+		It("install plugins and describe each installed plugin", func() {
+			for _, plugin := range framework.PluginsForLifeCycleTests {
+				target := plugin.Target
+				if plugin.Target == framework.GlobalTarget { // currently target "global" is not supported as target for install command
+					target = ""
+				}
+				err := tf.PluginCmd.InstallPlugin(plugin.Name, target, plugin.Version)
+				Expect(err).To(BeNil(), "should not get any error for plugin install")
+				str, err := tf.PluginCmd.DescribePlugin(plugin.Name, plugin.Target)
+				Expect(err).To(BeNil(), "should not get any error for plugin describe")
+				Expect(str).NotTo(BeNil(), "there should be output for plugin describe")
+			}
+		})
+		// Test case: (negative) describe plugin with incorrect target type
+		It("plugin describe: describe installed plugin with incorrect target type", func() {
+			str, err := tf.PluginCmd.DescribePlugin(framework.PluginsForLifeCycleTests[0].Name, framework.RandomString(5))
+			Expect(str).To(BeEmpty(), "stdout should be empty when target type is incorrect for plugin describe")
+			Expect(err.Error()).To(ContainSubstring(framework.InvalidTargetSpecified))
+		})
+		// Test case: (negative) describe plugin with incorrect plugin name
+		It("plugin describe: describe installed plugin with incorrect plugin name as input", func() {
+			name := framework.RandomString(5)
+			str, err := tf.PluginCmd.DescribePlugin(name, framework.PluginsForLifeCycleTests[0].Target)
+			Expect(str).To(BeEmpty(), "stdout should be empty when plugin name is incorrect for plugin describe command")
+			Expect(err.Error()).To(ContainSubstring(fmt.Sprintf(framework.UnableToFindPlugin, name)))
+		})
+		// Test case: list plugin and validate number of plugins installed are same number of plugins installed in previous step
+		It("list plugins and check number plugins should be same as installed in previous test", func() {
+			pluginsList, err := tf.PluginCmd.ListPlugins()
+			Expect(err).To(BeNil(), "should not get any error for plugin list")
+			Expect(len(pluginsList)).Should(Equal(len(framework.PluginsForLifeCycleTests)), "plugins list should return all installed plugins")
+		})
```

Comment:
There are more things to verify in the list output.
We should be verifying that all the fields of the plugins are correct. E.g. `context`, `version`, `target`, and `status`

Reply:
Updated